.github/workflows/release.yaml

```diff
@@ -0,0 +1,35 @@
+name: Release
+on:
+  workflow_dispatch:
+    inputs:
+      next-version:
+        type: choice
+        options:
+          - bump-minor
+          - bump-patch
+
+
+jobs:
+  build:
+    uses: ./.github/workflows/build.yaml
+    permissions:
+      contents: write
+      id-token: write
+      packages: write
+    with:
+      mode: release
+
+  release-to-github-and-bump:
+    uses: gardener/cc-utils/.github/workflows/release.yaml@master
+    needs:
+      - build
+    secrets: inherit
+    permissions:
+      contents: write
+      id-token: write
+      packages: write
+    with:
+      release-commit-target: branch
+      next-version: ${{ inputs.next-version }}
```

Comment:
Out of curiosity, where does the `next-version` input come from? I guess it somehow depends on whether we are calling this from the main branch or from a hotfix release branch.

Reply:
this will be passed-in from the triggering-event. Depending on how you trigger the workflow, this will either be asked via a triggering-dialogue in GitHub-Actions-Web-UI, or you need to specify it via GitHub-API / GH-CLI.

Different to Concourse, GitHub-Actions _does_ allow passing-in values to a pipeline, so I figured to use this option. On the plus-side, this means to can explicitly choose this for each release (rather, as done in concourse, having to statically configure this, and have to generate pipelines w/ derived, hardcoded values). This comes at the (I hope neglegible) cost of having to to two extra-clicks for each release, if using the Web-UI.